backend/sources/views.py

```diff
@@ -317,12 +317,14 @@ def store(self, source_file, source_id, source_language, author, title):
         xml_sanitized_text = invalid_xml_remove(raw_text)
         text = html.escape(xml_sanitized_text)
         es.index(settings.ES_ALIASNAME, {
-            'id': source_id,
-            'language': source_language,
-            'author': author,
-            'title': title,
-            'text': text,
-            'text_{}'.format(source_language): text
+            **{
+                'id': source_id,
+                'language': source_language,
+                'author': author,
+                'title': title,
+                'text': text,
+            },
+            **({'text_{}'.format(source_language): text} if source_language != 'other' else {})
         })
```

Comment:
Here again the magical notation issue. Maybe you can use the same function as in `sources_to_elasticsearch.py`, if instead of a `serial` parameter, you add a `bare` parameter with a `dict` that already has all fields except for `text_{}`. The only thing the function needs to do, is to conditionally add the key and otherwise return the `dict` unmodified. Suppose the function is like this:

```py
def optional_localized(document):
    language = document['language']
    if language is not 'other':
        document['text_{}'.format(language)] = document['text']
    return document
```

Then the above snippet could turn into this:

```py
        es.index(settings.ES_ALIASNAME, optional_localized({
             'id': source_id,
             'language': source_language,
             'author': author,
             'title': title,
             'text': text,
        }))
```

And similar in the other module.